src/main/java/edu/kis/powp/jobs2d/command/gui/CommandManagerWindow.java

```diff
@@ -108,5 +137,37 @@ public void HideIfVisibleAndShowIfHidden() {
             this.setVisible(true);
         }
     }
-
+    public void drawCommands(Graphics g) {
+        int startPosX = 0, startPosY = 0, endPosX = 0, endPosY = 0;
+        g.translate(150, 150);
+        DriverCommand currentCommand = commandManager.getCurrentCommand();
+        if (currentCommand instanceof ComplexCommand) {
+            List<DriverCommand> commandsList = ((ComplexCommand) currentCommand).getListOfCommands();
+            for (DriverCommand cmd : commandsList) {
+                if (cmd instanceof SetPositionCommand) {
+                    startPosX = ((SetPositionCommand) cmd).getPosX();
+                    startPosY = ((SetPositionCommand) cmd).getPosY();
+                }
+                if (cmd instanceof OperateToCommand) {
```

Comment:
This is not an OO approach (instanceof). If you want avoid using Jobs2dDrivers, I suggest implementing a visitor drawing a preview. (see #13).

Reply:
Two new visitor classes have been created